mod_regression/controllers.py

```diff
@@ -62,9 +62,39 @@ def test_view(regression_id):
 
 
 @mod_regression.route('/test/<regression_id>/delete')
+@template_renderer()
+@check_access_rights([Role.contributor, Role.admin]) # confirmation to delete regression test
 def test_delete(regression_id):
-    # Delete the regression test
-    pass
+    """
+    Function to delete regression tests
+    :param regression_id: Id of the regression test to be deleted
+    :type regression_id: int
+    """
+
+    regression_test = RegressionTest.query.filter(RegressionTest.id == regression_id).first()  # find regression test corresponding to id given in argument
+
+    if regression_test is None:
+        abort(404) # aborting if a regression test with the given id doesn't exist
+
+    customized_tests = CustomizedTest.query.filter(CustomizedTest.regression_id == regression_id).all() # finding all customized tests that have the given regression test
+    regression_test_outputs = RegressionTestOutput.query.filter(RegressionTestOutput.regression_id == regression_id).all() # finding all regression test outputs that have the given regression test
+    test_results = TestResult.query.filter(TestResult.regression_test_id == regression_id).all() # finding all test results that have the given regression test
+    test_result_files = TestResultFile.query.filter(TestResultFile.regression_test_id == regression_id).all() # finding all test result files that have the given regression test
+
+    # Deleting the tests/files which have the given regression test
+
+    for customized_test in customized_tests:
```

Comment:
Check the models to see if this needs to be done explicitely, or if it's implicitely deleted by the relational models.